modules/context/auth.go

```diff
@@ -42,7 +42,7 @@ func Toggle(options *ToggleOptions) macaron.Handler {
 			// prevent infinite redirection
 			// also make sure that the form cannot be accessed by
 			// users who don't need this
-			if ctx.Req.URL.Path == setting.AppSubURL+"/user/settings/change_password" {
+			if ctx.Req.URL.Path == "/user/settings/change_password" {
```

Comment:
@techknowlogick  @lafriks - As described in the issue #5815 I do use `https://example.com/git` since we have **NOT** got a subdomain. And I did **test** it before submitting the PR.

So, as far as I could see the reverse proxy (`nginx`) removes `https://exapmle.com/git` form the request. maybe you might prefer the weird version:

```go
if setting.AppSubURL+ctx.Req.URL.Path == setting.AppSubURL+"/user/settings/change_password" {
```
Why? 

[setting.go:446](https://github.com/go-gitea/gitea/blob/master/routers/repo/setting.go#L446) -> ``setting.AppSubURL+ctx.Req.URL.Path`` is used as well....

Reply:
Oh, ok than your solution is fine, taking my comment back :)

Reply:
Similarly